docs/source/user-guide/sql/datafusion-functions.md

```diff
@@ -84,3 +84,31 @@ Note that `CAST(.. AS Timestamp)` converts to Timestamps with Nanosecond resolut
 - Other Timestamp() columns or values
 
 Note that `CAST(.. AS Timestamp)` converts to Timestamps with Nanosecond resolution; this function is the only way to convert/cast to seconds resolution.
+
+## `EXTRACT, date_part`
+
+`EXTRACT(field FROM source)`
+
+- The extract function retrieves subfields such as year or hour from date/time values.
+  source must be a value expression of type timestamp, Data32, Data64 field is an identifier that selects what field to extract from the source value.
+  The extract function returns values of type u32.
```

Comment:
I think the result type of u32 is not enough.
From the spark, 
```
spark-sql> SELECT EXTRACT(YEAR FROM TIMESTAMP '-2000-12-16 12:21:13');
-2000
```
But this is not related to this issue, I will open a follow up pull request and discuss the returned type.

Reply:
https://github.com/apache/arrow-datafusion/issues/1992

Reply:
@liukun4515 We need  a discussion  about wether need support negative `timestamp `or `date`

Reply:
I think this is just a little issue for me, after my careful research I will give a feedback in the issue